staging/src/k8s.io/apiextensions-apiserver/test/integration/fixtures/webhook.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fixtures
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+// WaitReadyFunc calls triggerConversionFn periodically and waits until it detects that the webhook
+// conversion server has handled at least 1 conversion request or the timeout is exceeded, in which
+// case an error is returned.
+type WaitReadyFunc func(timeout time.Duration, triggerConversionFn func() error) error
+
+// StartConversionWebhookServerWithWaitReady starts an http server with the provided handler and returns the WebhookClientConfig
+// needed to configure a CRD to use this conversion webhook as its converter.
+// It also returns a WaitReadyFunc to be called after the CRD is configured to wait until the conversion webhook handler
+// accepts at least one conversion request. If the server fails to start, an error is returned.
+// WaitReady is useful when changing the conversion webhook config of an existing CRD because the update does not take effect immediately.
+func StartConversionWebhookServerWithWaitReady(handler http.Handler) (func(), *apiextensionsv1beta1.WebhookClientConfig, WaitReadyFunc, error) {
```

Comment:
all this should go into one package. If rebased onto https://github.com/kubernetes/kubernetes/pull/77868, test/integration/convert is a good spot.

Reply:
Why do we need to expose the wait function at all? Would hide that.

Reply:
Sounds good, I'll rebase shortly.